### PR TITLE
[warmboot] Add workaround for `INIT_VIEW` failure 

### PR DESCRIPTION
**Pull request by zjswhhh**
**MICROSOFT ADO#: 24107886**

## why change is needed
"SaiGetHardwareInfo" is now returning string "none" instead of empty string (""), which is causing INIT_VIEW failure:
```
May 26 18:57:33.577868 NOTICE syncd#syncd: :- onSwitchCreateInInitViewMode: new switch oid:0x21000000000000 contains hardware info: ''
May 26 18:57:33.577868 ERR syncd#syncd: :- onSwitchCreateInInitViewMode: hardware info mismatch: current 'none' vs new ''
May 26 18:57:33.578017 ERR syncd#syncd: :- run: Runtime error: :- onSwitchCreateInInitViewMode: hardware info mismatch: current 'none' vs new ''
```

Adding this change as a temp workaround before CSP is resolved with BRCM. 

## how I verify 
Test syncd build on lab device, able to pass INIT_VIEW stage. 

sign-off: Jing Zhang zhangjing@microsoft.com

### CODE REVIEW

File: syncd/Syncd.cpp
Comment (vaibhavhd) [inline syncd/Syncd.cpp:4312]: Please add a comment here explaining why this is needed, and that the behavior change is from SAI call.
Also a TODO to remove this change in the future if SAI vendor agrees to fix this at their end.

**Review by vaibhavhd — DISMISSED** [review]
LGTM, please wait for @kcudnik 's review. Thanks.

File: syncd/Syncd.cpp
Comment (zjswhhh) [reply syncd/Syncd.cpp:4317]: Done.

**Review by saiarcot895 — DISMISSED** [review]

File: syncd/Syncd.cpp
Comment (yxieca) [inline syncd/Syncd.cpp:4317]: The coding style in this repo seems to require white space before and after ":".

Comment (kcudnik) [reply syncd/Syncd.cpp:4317]: How come suddenly its none? Is it Sai internal vendor?

Comment (kcudnik) [reply syncd/Syncd.cpp:4317]: You miss space on last comment linę before star

Comment (vaibhavhd) [reply syncd/Syncd.cpp:4317]: > How come suddenly its none? Is it Sai internal vendor?

With new SAI code in 202205+ images the return value for GET call on hwinfo is `none`. We have confirmed that older SAI was returning `''`. This will be further  discussed with BRCM SAI team. But, in the meanwhile, we need this workaround to unblock/uncover other issues.

**Review by kcudnik — APPROVED** [review]

File: syncd/Syncd.cpp
Comment (zjswhhh) [reply syncd/Syncd.cpp]: @yxieca @kcudnik Addressed.

**Review by vaibhavhd — APPROVED** [review]

**Review by yxieca — APPROVED** [review]

**Pull request merged**
